packages/expo-audio/android/src/main/java/expo/modules/audio/AudioPlayer.kt

```diff
@@ -104,6 +136,16 @@ class AudioPlayer(
       emit("onPlaybackStatusUpdate", body)
     }
 
+  private suspend fun sendAudioSampleUpdate(sample: List<Float>) = withContext(Dispatchers.Main) {
```

Comment:
it looks like main thread is not necessary? we just doing some calculation and emit to js thread here.